File: src/me/krickl/memebotj/Cooldown.java
package me.krickl.memebotj;

public class Cooldown {
	private int cooldownLen = 0;
	private int cooldownStart = 0;
	private int cooldownEnd = 0;

	public Cooldown(int cooldownLen) {
		this.cooldownLen = cooldownLen;
	}

	public void startCooldown() {
		this.cooldownStart = (int) (System.currentTimeMillis() / 1000L);
		this.cooldownEnd = (int) (System.currentTimeMillis() / 1000L) + this.cooldownLen;
	}

	public int getCooldownLen() {
		return cooldownLen;
	}

	public void setCooldownLen(int cooldownLen) {
		this.cooldownLen = cooldownLen;
	}

	public void setCooldownEnd(int cooldownEnd) {
		this.cooldownEnd = cooldownEnd;
	}

	public boolean canContinue() {
		if (this.cooldownEnd > (int) (System.currentTimeMillis() / 1000L)) {
			return false;
		}

		return true;
	}
}

File: src/me/krickl/memebotj/Cooldown.scala
package me.krickl.memebotj

import scala.beans.BeanProperty
;

class Cooldown(val cdLen: Integer) {
	@BeanProperty
	var cooldownLen: Integer = cdLen
	@BeanProperty
	var cooldownStart = 0
	@BeanProperty
	var cooldownEnd = 0

	def startCooldown() = {
		this.cooldownStart = (System.currentTimeMillis() / 1000).toInt
		this.cooldownEnd = (System.currentTimeMillis() / 1000L).toInt + this.cooldownLen
	}

	def canContinue: Boolean = {
		if (this.cooldownEnd > (System.currentTimeMillis() / 1000).toInt) {
			return false
		}

    true
	}
}
